### PR TITLE
orocksdb.0.2.1

### PR DESCRIPTION
**Pull request by domsj**

### CODE REVIEW

**Comment by domsj** [comment]
Is there a way to get the contents of https://gist.githubusercontent.com/domsj/9a28ba5a523a3420ded8/raw/dc6ebd8768a87dee285e84438dc733dfc58e037f/gistfile1.txt expressed directly in the opam file?

Basically what I want to do is call the install script from my repo and pass it an env argument so that it will use g++-4.8 instead of the travis default g++ 4.6.

**Comment by domsj** [comment]
Build failure seems to be unrelated to my change?

**Comment by samoht** [comment]
> Is there a way to get the contents of https://gist.githubusercontent.com/domsj/9a28ba5a523a3420ded8/raw/dc6ebd8768a87dee285e84438dc733dfc58e037f/gistfile1.txt expressed directly in the opam file?


do you want to use that as a build instruction? You can always use `sh -c <unsafe things>` ...

File: packages/orocksdb/orocksdb.0.2.1/opam [hunk 1]
```diff
@@ -0,0 +1,28 @@
+opam-version: "1.2"
+name: "orocksdb"
+version: "0.2.1"
```
Comment (samoht) [inline packages/orocksdb/orocksdb.0.2.1/opam]: Can you remove the name and version from the opam file? Thanks!

Comment (domsj) [inline packages/orocksdb/orocksdb.0.2.1/opam]: Ok

**Comment by domsj** [comment]
Not sure I fully understand your question...

For the travis build here I want to specify it should use g++-4.8.
For the script in my repository, which people can also use without opam, I don't want to force them to have g++-4.8 available. (Clang or another g++ version -- as long as it support c++11 -- is fine too.)
Preferable I can do that without introducing the extra gist file or any other duplication...

(Actually it would be even better if I could just tell opam: I need a c++11 compiler, make sure it's there and that it is the default when compiling whatever my package wants to compile.)

Anyway, as far as I'm concerned this shouldn't stop us from integrating this package. I'm just looking for better (lazier) ways to do this in the future...

**Comment by avsm** [comment]
So there are two ways to do this:

- create a `conf-c++11` package that performs the necessary system probing, and depend on that.
- Extend the horrible hack in the opam-repostiory travis script:

```
    case $TRAVIS_OS_NAME in
        linux)
            # we need fresh gcc and binutils, maybe...
            # this can soon be removed, once travis upgraded their infrastructure
            if [ `opam install --dry-run $pkg | grep -c mirage-entropy-xen` -gt 0 ] ; then
                echo "installing a recent gcc and binutils (mainly to get mirage-entropy-xen working\!)"
                sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/test
                sudo apt-get -qq update
                sudo apt-get install -y gcc-4.8
                sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.8 90
                wget http://mirrors.kernel.org/ubuntu/pool/main/b/binutils/binutils_2.24-5ubuntu3.1_amd64.deb
                sudo dpkg -i binutils_2.24-5ubuntu3.1_amd64.deb
            fi
    esac
```

To add more than the mirage--entropy-xen package.

File: packages/orocksdb/orocksdb.0.2.1/opam [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+build: [
+  [make]
+]
+install: [make "install"]
+remove: ["ocamlfind" "remove" "rocks"]
+depends: [
+  "ocamlfind" {build}
+  "ctypes" { >= "0.4.0" }
+  "ctypes-foreign" { >= "0.4.0"}
+  "depext" {build}
```
Comment (avsm) [inline packages/orocksdb/orocksdb.0.2.1/opam]: Is the `depext` dependency really needed for build? It should autoinstall if invoked (when using OPAM 1.2.2 or higher)

Comment (domsj) [inline packages/orocksdb/orocksdb.0.2.1/opam]: I'll leave it out next time (if I don't forget)

**Pull request merged**